crates/services/p2p/src/service.rs

```diff
@@ -1044,6 +1060,31 @@ impl SharedState {
         Ok((peer_id.to_bytes(), data))
     }
 
+    pub async fn get_transactions(
+        &self,
+        range: Range<u32>,
+    ) -> anyhow::Result<(Vec<u8>, Option<Vec<Transactions>>)> {
+        let (sender, receiver) = oneshot::channel();
+
+        if range.is_empty() {
+            return Err(anyhow!(
+                "Cannot retrieve transactions for an empty range of block heights"
+            ));
+        }
+
+        self.request_sender
+            .send(TaskRequest::GetTransactions {
+                block_height_range: range,
+                channel: sender,
+            })
+            .await?;
```

Comment:
perhaps we should continue to use the `TaskRequest::GetTransactionsFromPeer` and perform peer selection for a given block height instead - https://github.com/FuelLabs/fuel-core/blob/f597ac9db0602caed7e20eaeafed0a1bd2e96ca7/crates/services/p2p/src/service.rs#L264-L266

Reply:
I personally think that there is still use case where it's interesting to already give a peer that we know have the information, we use this un sync service.

Reply:
right, i agree ~ but this information is also valid for a peer returned by `get_peer_id_with_height`. perhaps we should not duplicate valid peers for a given height then.

Reply:
I don't get your comment. My flow is that I ask for an header and so the peer is returned along with the header then I can directly ask to this peer about the transactions because in our network if you have the header we assume you have the transactions. It avoid running this `get_peer_id_with_height` function that can be costly.
Maybe I miss your point and in this case we can go over more voice discussion :)

Reply:
> perhaps we should continue to use the TaskRequest::GetTransactionsFromPeer and perform peer selection for a given block height instead

If we do peer selection outside of the `run` loop, we can face race condition were peer is disconnected already